hotel_reservation/models/hotel_reservation.py

```diff
@@ -0,0 +1,1240 @@
+# See LICENSE file for full copyright and licensing details.
+
+import time
+from datetime import datetime, timedelta
+
+import pytz
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT as dt
+
+
+class HotelFolio(models.Model):
+
+    _inherit = "hotel.folio"
+    _order = "reservation_id desc"
+
+    reservation_id = fields.Many2one("hotel.reservation", "Reservation Id")
+
+    def write(self, vals):
+        context = dict(self._context)
+        if not context:
+            context = {}
+        context.update({"from_reservation": True})
+        res = super(HotelFolio, self).write(vals)
+        reservation_line_obj = self.env["hotel.room.reservation.line"]
+        for folio in self:
+            for reservation in folio.reservation_id:
+                reservation_obj = reservation_line_obj.search(
+                    [("reservation_id", "=", reservation.id)]
+                )
+                if len(reservation_obj) == 1:
+                    for line_id in reservation.reservation_line:
+                        line_id = line_id.reserve
+                        for room_id in line_id:
+                            vals = {
+                                "room_id": room_id.id,
+                                "check_in": folio.checkin_date,
+                                "check_out": folio.checkout_date,
+                                "state": "assigned",
+                                "reservation_id": reservation.id,
+                            }
+                            reservation_obj.write(vals)
```

Comment:
Would be nice to have to use like as line/line_rec instead of line_id because it's an object.

Reply:
Done.